apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -207,6 +219,54 @@ type TopologySpreadConstraint struct {
 	WhenUnsatisfiable UnsatisfiableConstraintAction `json:"whenUnsatisfiable,omitempty"`
 }
 
+// RolloutStrategy describes how to update existing placed resource either onto a different cluster
+// or in-place upgrade to the latest resource.
+type RolloutStrategy struct {
+	// Type of rollout. The only supported type is "RollingUpdate". Default is "RollingUpdate".
+	// +optional
+	Type RolloutStrategyType `json:"type,omitempty"`
+
+	// Rolling update config params. Present only if RolloutStrategyType = RollingUpdate.
+	// +optional
+	RollingUpdate *RollingUpdateConfig `json:"rollingUpdate,omitempty"`
+}
+
+// +enum
+type RolloutStrategyType string
+
+const (
+	// RollingUpdateRolloutStrategyType replace the old placed resource using rolling update
+	// i.e. gradually create the new one while replace the old ones.
+	RollingUpdateRolloutStrategyType RolloutStrategyType = "RollingUpdate"
+)
+
+// RollingUpdateConfig contains the config to control the desired behavior of rolling update.
+type RollingUpdateConfig struct {
+	// The maximum number of clusters that can be unavailable during the update.
+	// Absolute number is calculated from percentage by applying to the new target number of clusters.
+	// We consider a resource no available when we either remove it from a cluster or in-place
+	// upgrade the resources content on the same cluster.
+	// This can not be 0 if MaxSurge is 0.
```

Comment:
Hi Ryan! Just for a bit clarification here: what would the behavior be for downscaling when maxSurge is > 0 but maxUnavailable is set to 0?

Reply:
I don't think maxUnavailable applies when it's down scaling.

Reply:
Got it 👌